README.adoc

```diff
@@ -198,16 +208,142 @@ curl http://localhost:9080/hello
 
-To stop and remove your container, run the following commands:
+InstantOn requires a Linux system with kernel version 5.9 or greater. If your system does not meet this prerequisite, go to the <<running-the-application-on-open-liberty>> section. You can run the following command to check out the kernel version of your system:
+
+[role='command']
+```
+uname -r
+```
+
+The https://openliberty.io/docs/latest/reference/feature/crac.html[Coordinated Restore at Checkpoint^] (`crac`) feature enables applications to use the Liberty InstantOn implementation of the https://javadoc.io/doc/org.crac/crac/1.4.0/index.html[org.crac^] APIs. The Spring Framework hooks into the checkpoint and restore when the `org.crac` APIs are enabled.
+
+[role="code_command hotspot file=0", subs="quotes"]
+----
+#Replace the `Dockerfile` in the `start` directory.#
+`Dockerfile`
+----
+
+Dockerfile
+[source, Text, linenums, role='code_column']
+----
+include::instantOn/Dockerfile[]
+----
+
+crac.xml
+[source, xml, linenums, role='code_column']
+----
+include::finish/src/main/liberty/instantOn/crac.xml[]
+----
+
+To enable the `crac` feature, add the [hotspot=cracXml file=0]`COPY` command that copies the provided `crac.xml` configuration file to the Liberty `configDropins` configuration directory. The `crac.xml` file enables the [hotspot=crac file=1]`crac` feature to the Liberty instance.
+
+Run the following command to rebuild the Docker image:
+[role='command']
+```
+docker build -t springboot .
+```
+
+To take a checkpoint of the application process with the `afterAppStart` option, run the following command:
+
+[role='command']
+```
+docker run \
+  --name springBootCheckpointContainer \
+  --privileged \
+  --env WLP_CHECKPOINT=afterAppStart \
+  springboot
+```
+
+When the application process checkpoint completes, the `springBootCheckpointContainer` application container is stopped and exits. The stopped `springBootCheckpointContainer` container contains the data from the InstantOn checkpoint process. To take this checkpoint process data and commit it to an application container image layer, run the following commands:
+
+[role='command']
+```
+docker commit springBootCheckpointContainer springboot-instanton
+docker rm springBootCheckpointContainer
+```
+
+The stopped `springBootCheckpointContainer` container is no longer needed and can safely be removed.
+
+To verify that the InstantOn image is built, run the `docker images` command to list all local Docker images:
+
+[role='command']
+```
+docker images
+```
+
+Your `springboot-instanton` image appears in the list of Docker images. It shows a faster startup time than the original `springboot` image.
```

Comment:
Is this true? The docker images list just shows that the images were created 20 secs vs 14 mins ago. The latter just means that the container was created 14 mins ago, not that it took 14 mins to startup, surely?

Reply:
If it means in 20 secs in comparison to the 27 secs in the earlier docker images list, I think that needs to be explicitly stated in the text, not just left to the reader to work out the comparison for themself.

Reply:
Also this isn't really a good way to measure startup time, is it? Isn't it better to highlight how long it took to start by comparing Liberty terminal startup output messages?

Reply:
Should the comparison (a few lines down from here, rather than here) be 5.542 seconds vs 0.585 seconds ?

Reply:
Instead of showing the comparison of 5.542 seconds vs 0.585 seconds, the guide just says "Liberty’s startup is complete in less than a second:". Users may not remember the exact seconds for the original `springboot` image. Also, I prefer not hard-code the values in any statements, in the output as an example is fine.

Reply:
I moved the "It shows ..."  statement after "Liberty’s startup is complete ..."